### ISSUE TITLE
Some Inquires About Arabic Finetuning.

### ISSUE THREAD

**Issue by MohieEldinMuhammad** [PaddlePaddle/PaddleOCR#7623]
My goal is to finetune the Arabic model:
 ```https://paddleocr.bj.bcebos.com/PP-OCRv3/multilingual/arabic_PP-OCRv3_rec_train.tar``` on **sentence-level**
This is a sample from the multilingual dataset you provided:
![image](https://user-images.githubusercontent.com/89604782/190574488-b3bce7cb-2ca5-4415-b5e3-1575e631f65c.png)

**Question 1)** If I will fine-tune images with both Arabic and English will this work or it's recommended to use images with only Arabic language? How the **dict.txt** will look like if I will train on both English and Arabic?

**Question 2)** should I use the whole image with multiple lines like this or crop the image to contain only one text line in the image like this? 
![image](https://user-images.githubusercontent.com/89604782/190575890-7eb9cf5e-7cc8-4794-a801-6f192a4cc501.png)

**Question 3)** given the ground truth of this sample image it's made on word-level, If I will train on sentence-level should I do some manipulation to merge all words together and then merge .txt files in one .txt file named **rec_gt_train.txt**?

**Question 4)** I'm afraid of the prediction direction while training to have opposite direction with the ground truth text, how the matching between prediction text and ground truth text will be calculated?

**Comment by andyjpaddle** [comment]
refer this doc https://github.com/PaddlePaddle/PaddleOCR/blob/release/2.6/doc/doc_en/multi_languages_en.md
1. the Arabic model is trained using a dict which include english and Arabic, the dict is https://github.com/PaddlePaddle/PaddleOCR/blob/release/2.6/ppocr/utils/dict/arabic_dict.txt
2. for text recognition model, you should prepare one text line image
3. for each image, it's gt is a line word, supported space char. One image, one gt
4. the latest code have fixed it

**Comment by MohieEldinMuhammad** [comment]
I should support the white space as a character or not ?
Also for Detection the model detect mostly word by word, can finetuning on sentences fix this ?
Example:
![image](https://user-images.githubusercontent.com/89604782/190639432-fe9cec6f-19c1-4253-8df2-12ca872d3676.png)

**Comment by MohieEldinMuhammad** [comment]
Also when making a .txt label file for each image dose ```SimpleDataSet```  will read it normally or i need to make some edits to read the ground truth file by file ?

**Comment by andyjpaddle** [comment]
> I should support the white space as a character or not ? Also for Detection the model detect mostly word by word, can finetuning on sentences fix this ? Example: ![image](https://user-images.githubusercontent.com/89604782/190639432-fe9cec6f-19c1-4253-8df2-12ca872d3676.png)

space char is supported if `use_space_char` is set true
you can increase the `unclip_ratio` param or you can union this bbox by own postprocess

**Comment by andyjpaddle** [comment]
> I should support the white space as a character or not ? Also for Detection the model detect mostly word by word, can finetuning on sentences fix this ? Example: ![image](https://user-images.githubusercontent.com/89604782/190639432-fe9cec6f-19c1-4253-8df2-12ca872d3676.png)

space char is supported if `use_space_char` is set true
you can increase the `unclip_ratio` param or you can union this bbox by own postprocess, finetune also can fix this

**Comment by andyjpaddle** [comment]
> Also when making a .txt label file for each image dose `SimpleDataSet` will read it normally or i need to make some edits to read the ground truth file by file ?

combine all .txt to one txt file

**Comment by MohieEldinMuhammad** [comment]
thanks

**Comment by andyjpaddle** [comment]
Because it's a two stage task, detection model is responsible for detect the text location and recognition model is responsible for recognize what the text line is. 
Also you can use the end2end method(Eg: PGNet) to takes whole image and output all sentences in it.
Usually the two stage method has a good performance and has a little model size. end2end method is  harder than two stage for now.

**Comment by MohieEldinMuhammad** [comment]
Thanks for reply,

so this will be the dir of the data:
```
trian_data
  train
  test
  train_gt
  test_gt
```
where gt is:
```
image_1_sen1 example sentence
image_1_sen2 example sentence 
...

**Comment by MohieEldinMuhammad** [comment]
i also want the most accurate model even if it's slower so i was checking the combination of the models with the backbones, but i didn't find the combination of the model and backbone that is in the arabic config file
![image](https://user-images.githubusercontent.com/89604782/194700414-6368ce6d-2875-4ca1-a259-888899244660.png)

![image](https://user-images.githubusercontent.com/89604782/194700528-3e93cc00-f92d-4f87-ba22-3fbc2881e9bf.png)

you see --> there is no SVTR with MobileNet backbone so if i want the most accurate combination what should i choose ?

**Comment by andyjpaddle** [comment]
> Thanks for reply,
> 
> so this will be the dir of the data:
> 
> ```
> trian_data
>   train
>   test
>   train_gt
>   test_gt
> ```
> 
> where gt is:
> 
> ```
> image_1_sen1 example sentence
> image_1_sen2 example sentence 
> ... 
> ```

please read this doc https://github.com/PaddlePaddle/PaddleOCR/blob/release/2.6/doc/doc_en/dataset/ocr_datasets_en.md#2-text-recognition

**Comment by andyjpaddle** [comment]
> i also want the most accurate model even if it's slower so i was checking the combination of the models with the backbones, but i didn't find the combination of the model and backbone that is in the arabic config file ![image](https://user-images.githubusercontent.com/89604782/194700414-6368ce6d-2875-4ca1-a259-888899244660.png)
> 
> ![image](https://user-images.githubusercontent.com/89604782/194700528-3e93cc00-f92d-4f87-ba22-3fbc2881e9bf.png)
> 
> you see --> there is no SVTR with MobileNet backbone so if i want the most accurate combination what should i choose ?

SVTR with MobileNet backbone is our proposed PP-OCRv3 rec model. you can refer to this doc https://github.com/PaddlePaddle/PaddleOCR/blob/release/2.6/doc/doc_en/PP-OCRv3_introduction_en.md
https://github.com/PaddlePaddle/PaddleOCR/blob/release/2.6/doc/doc_en/algorithm_overview_en.md#12-text-recognition-algorithms These model are from the open-source paper, we just only reproduce this using paddle which has no improvement on them.

**Comment by andyjpaddle** [comment]
I suggest you read these docs firstly

**Comment by MohieEldinMuhammad** [comment]
``` 
please read this doc https://github.com/PaddlePaddle/PaddleOCR/blob/release/2.6/doc/doc_en/dataset/ocr_datasets_en.md#2-text-recognition
```
yes i read it multiple times but it's made on word level not on sentences level that's why I'm making sure from the ground truth shape. I've tried to train with a separate gt for each image but got errors so I should combine all sentences images in one gt just like words gt right ?

**Comment by andyjpaddle** [comment]
one image one line word, space is also supported
but if the two words is far away, more than one space, you should crop them into two images. 
Two line words in an image is not supported

**Comment by MohieEldinMuhammad** [comment]
> I suggest you read these docs firstly

I was just confused that you are linking SVTR with SVTR-Tiny backbone but the arabic model it has Mobile-Net which i couldn't find any accuracy for this combination.
so you mean to experiment multiple models and backbone and see the results by self no reference to the best accuracy ?

**Comment by MohieEldinMuhammad** [comment]
> one image one line word, space is also supported but if the two words is far away, more than one space, you should crop them into two images. Two line words in an image is not supported

and for ground truth, it will be **one** with **all** the images paths and text right ?

**Comment by andyjpaddle** [comment]
gt is a txt file, including all the data pairs (image path   label)




发自我的手机


-------- 原始邮件 --------
发件人： MohieEldinMuhammad ***@***.***>
日期： 2022年10月8日周六 傍晚5:54
收件人： PaddlePaddle/PaddleOCR ***@***.***>
抄送： "Jiang,Kaitao" ***@***.***>, Mention ***@***.***>
主 题： Re: [PaddlePaddle/PaddleOCR] Some Inquires About Arabic Finetuning. (Issue #7623)

one image one line word, space is also supported but if the two words is far away, more than one space, you should crop them into two images. Two line words in an image is not supported

and for ground truth, it will be one with all the images paths and text right ?

―

**Comment by MohieEldinMuhammad** [comment]
Thanks a lot, and sorry for asking many questions

**Comment by andyjpaddle** [comment]
docs can help you a lot, source-code can further help you, just need time to read
